libs/cardano-ledger-core/testlib/Test/Cardano/Ledger/Core/Binary/CDDL.hs

```diff
@@ -419,3 +463,224 @@ mkScriptInvalidHereafter =
+
+ipv4 :: Rule
+ipv4 = "ipv4" =:= VBytes `sized` (4 :: Word64)
+
+ipv6 :: Rule
+ipv6 = "ipv6" =:= VBytes `sized` (16 :: Word64)
+
+dns_name64 :: Rule
+dns_name64 = "dns_name" =:= VText `sized` (0 :: Word64, 64 :: Word64)
+
+dns_name128 :: Rule
+dns_name128 = "dns_name" =:= VText `sized` (0 :: Word64, 128 :: Word64)
+
+url64 :: Rule
+url64 = "url" =:= VText `sized` (0 :: Word64, 64 :: Word64)
+
+url128 :: Rule
+url128 = "url" =:= VText `sized` (0 :: Word64, 128 :: Word64)
+
+single_host_addr :: Named Group
+single_host_addr =
+  "single_host_addr"
+    =:~ grp
+      [ 0
+      , a $ port / VNil
+      , a $ ipv4 / VNil
+      , a $ ipv6 / VNil
+      ]
+
+drep :: Rule
+drep =
+  "drep"
+    =:= arr [0, a addr_keyhash]
+    / arr [1, a script_hash]
+    / arr [2] -- always abstain
+    / arr [3] -- always no confidence
+
+anchor :: Rule
+anchor = "anchor" =:= arr ["anchor_url" ==> url128, "anchor_data_hash" ==> hash32]
+
+stake_registration_cert_deprecated :: Named Group
+stake_registration_cert_deprecated =
+  comment "This certificate will be deprecated in a future era" $
+    "stake_registration_cert_deprecated" =:~ grp [0, a stake_credential]
+
+stake_registration_cert :: Named Group
+stake_registration_cert = "stake_registration_cert" =:~ grp [7, a stake_credential, a coin]
+
+drep_registration_cert :: Named Group
+drep_registration_cert = "drep_registration_cert" =:~ grp [16, a drep_credential, a coin, a $ anchor / VNil]
+
+stake_deregistration_cert_deprecated :: Named Group
+stake_deregistration_cert_deprecated =
+  comment "This certificate will be deprecated in a future era" $
+    "stake_deregistration_cert_deprecated" =:~ grp [1, a stake_credential]
+
+stake_deregistration_cert :: Named Group
+stake_deregistration_cert = "stake_deregistration_cert" =:~ grp [8, a stake_credential, a coin]
+
+drep_deregistration_cert :: Named Group
+drep_deregistration_cert = "drep_deregistration_cert" =:~ grp [17, a drep_credential, a coin]
+
+stake_delegation_cert :: Named Group
+stake_delegation_cert = "stake_delegation_cert" =:~ grp [2, a stake_credential, a pool_keyhash]
+
+vote_delegation_cert :: Named Group
+vote_delegation_cert = "vote_delegation_cert" =:~ grp [9, a stake_credential, a drep]
+
+stake_vote_delegation_cert :: Named Group
+stake_vote_delegation_cert =
+  "stake_vote_delegation_cert" =:~ grp [10, a stake_credential, a pool_keyhash, a drep]
+
+stake_registration_delegation_cert :: Named Group
+stake_registration_delegation_cert =
+  "stake_registration_delegation_cert" =:~ grp [11, a stake_credential, a pool_keyhash, a coin]
+
+vote_registration_delegation_cert :: Named Group
+vote_registration_delegation_cert =
+  "vote_registration_delegation_cert" =:~ grp [12, a stake_credential, a drep, a coin]
+
+stake_vote_registration_delegation_cert :: Named Group
+stake_vote_registration_delegation_cert =
+  "stake_vote_registration_delegation_cert"
+    =:~ grp [13, a stake_credential, a pool_keyhash, a drep, a coin]
```

Comment:
Instead of coming up with all those separate names, I think we can just group them together into an alternative, which will reduce cognitive overhead for the reader:
```suggestion
account_registration_delegation_cert :: Named Group
  "account_registration_delegation_cert" =:~ 
    grp [12, a stake_credential, a drep, a coin]
  \ grp [13, a stake_credential, a pool_keyhash, a drep, a coin]
  \ grp [14, a committee_cold_credential, a committee_hot_credential]
```